lib/src/foundation/yaru_entry_segment.dart

```diff
@@ -0,0 +1,458 @@
+    assert(maxLength == null || text.length <= maxLength!);
+    return text;
+  }
+
+  @override
+  int get length => text.length;
+
+  String? _input;
+  @override
+  String? get input => _input;
+
+  bool _selected = false;
+  bool _squashOnNextInput = false;
+
+  @override
+  void onSelect(bool selected) {
+    if (_selected == false && selected == true) {
+      _squashOnNextInput = true;
+    }
+    _selected = selected;
+  }
+
+  @override
+  YaruSegmentEventReturnAction onInput(String input) {
+    var action = YaruSegmentEventReturnAction.handled;
+    final previousInput = _input;
+
+    if (_squashOnNextInput) {
+      _input = null;
+      _squashOnNextInput = false;
+    }
+
+    _input = (_input ?? '') + input;
+
+    if (maxLength != null && _input!.length >= maxLength!) {
+      _input = _input!.getNbFirstCharacter(maxLength!);
+      action = YaruSegmentEventReturnAction.selectNextSegment;
+    }
+
+    if (_input != previousInput) notifyListeners();
+    return action;
+  }
+
+  @override
+  YaruSegmentEventReturnAction onBackspaceKey() {
+    if (_input != null) {
+      _input = null;
+      notifyListeners();
+      return YaruSegmentEventReturnAction.handled;
+    }
+
+    return YaruSegmentEventReturnAction.selectPreviousSegment;
+  }
+
+  @override
+  YaruSegmentEventReturnAction onDownArrowKey() =>
+      YaruSegmentEventReturnAction.ignored;
+
+  @override
+  YaruSegmentEventReturnAction onUpArrowKey() =>
+      YaruSegmentEventReturnAction.ignored;
+}
+
+typedef YaruNumericSegmentCallback = int? Function(
+  String? input,
+  int? value,
+  int? oldValue,
+)?;
+
+class YaruNumericSegment extends ChangeNotifier implements YaruEntrySegment {
+  /// Creates a [YaruNumericSegment].
+  YaruNumericSegment({
+    required this.minLength,
+    required this.maxLength,
+    int? initialValue,
+    required this.placeholderLetter,
+    this.onValueChange,
+    this.onInputCallback,
+    this.onDownArrowKeyCallback,
+    this.onUpArrowKeyCallback,
+  })  : assert(minLength > 0),
+        assert(maxLength == null || maxLength >= minLength),
+        _value = initialValue;
+
+  /// Creates a [YaruNumericSegment] with a fixed length.
+  YaruNumericSegment.fixed({
+    required int length,
+    int? initialValue,
+    required this.placeholderLetter,
+    this.onValueChange,
+    this.onInputCallback,
+    this.onDownArrowKeyCallback,
+    this.onUpArrowKeyCallback,
+  })  : assert(length > 0),
+        _value = initialValue,
+        minLength = length,
+        maxLength = length;
+
+  int? _value;
+  int? get value => _value;
+  set value(int? value) {
+    final oldValue = _value;
+
+    _value = onValueChange != null
+        ? onValueChange!(
+            _input,
+            value,
+            oldValue,
+          )
+        : value;
+    if (maxLength != null && _value != null && _value!.length > maxLength!) {
+      _value = oldValue;
+    }
+
+    if (_value != oldValue) notifyListeners();
```

Comment:
Should null values always notify? (Just asking because I often need to switch these decision in various situations)

Reply:
What do you mean exactly? I'm not sure to understand.
If the value was null and becomes int or was int and becomes null it should notify, isn't it?

Reply:
yes, was just asking if you want to allow to set it to null if it wasn't before :)

Reply:
Okay, so yeah this is intended 👌🙂